docs/tutorials/01_basics/04_strings.md

```diff
@@ -0,0 +1,302 @@
+message = "Crystal is awesome"
+
+p! message.index("s"),
+   message.index("s", offset: 4),
+   message.index("s", offset: 10)
+```
+
+The method `#rindex` works exactly the same, but instead it searches from the end of the string.
+
+```{.crystal .crystal-play}
+message = "Crystal is awesome"
+
+p! message.rindex("s"),
+   message.rindex("s", 13),
+   message.rindex("s", 8)
+```
+
+In case the substring is not found, the result is a special value called `nil`.
+It basically means "no value". Which makes sense when the substring has no index.
+
+Looking at the return type of `#index` we can see that it returns either `Int32` or `Nil`.
+
+```{.crystal .crystal-play}
+a = "Crystal is awesome".index("aw")
+p! a, typeof(a)
+b = "Crystal is awesome".index("meh")
+p! b, typeof(b)
+```
+
+!!! tip
+    We'll cover `nil` more deeply in the next lesson.
+
+## Extracting Substrings
+
+A substring is a part of a string. If you want to extract parts of the string,
+there are several ways to do that.
+
+The index accessor `#[]` allows referencing a substring by character index and size. Character
+indices start at `0` and reach to length (i.e. the value of `#size`) minus one.
+The first argument specifies the index of the first character that is supposed to be in the substring,
+and the second argument specifies the length of the substring. `message[6, 5]` extracts a substring
+of five characters long, starting at index six.
+
+```{.crystal .crystal-play}
+message = "Hello World!"
+
+p! message[6, 5]
+```
+
+Let's assume we have established that the string starts with `Hello` and ends with `!` and want to extract what's in
+between.
+If the message was `Hello Crystal`, we wouldn't get the entire word `Crystal` because it's longer than five characters.
+
+A solution is to calculate the length of the substring from the length of the entire string minus the lengths of beginning and end.
+
+```{.crystal .crystal-play}
+message = "Hello World!"
+
+p! message[6, message.size - 6 - 1]
+```
+
+There's an easier way to do that: The index accessor can be used with a [`Range`](https://crystal-lang.org/api/latest/Range.html)
+of character indices. A range literal consists of a start value and an end value, connected by two dots (`..`).
+The first value indicates the start index of the substring, as before, but the second is the end index (as opposed to the length).
+Now we don't need to repeat the start index in the calculation, because the end index is just the size minus two
+(one for the end index, and one for excluding the last character).
+
+It can be even easier: Negative index values automatically relate to the end of the string, so we don't need to calculate
+the end index from the string size explicitly.
+
+```{.crystal .crystal-play}
+message = "Hello World!"
+
+p! message[6..(message.size - 2)],
+   message[6..-2]
+```
+
+## Substitution
+
+In a very similar manner, we can alter a string. Let's make sure we properly greet Crystal and nothing else.
```

Comment:
Uh we literally can't alter a string. Maybe it's not important to stress that we *can't* but probably don't falsely say that we can.

Reply:
Hm, what would be a better term then?